js/src/utils/external.ts

```diff
@@ -21,36 +21,48 @@ export async function sendProcessReq(info: {
     );
     return true;
   }
+
   try {
-    // Spawn a child process to execute a Node.js script
+    // Use node-fetch for making HTTP requests
+    const url = new URL(info.url);
```

Comment:
The URL parsing should include error handling. Consider wrapping the URL constructor in a try-catch block to handle invalid URLs gracefully:
```typescript
let url;
try {
  url = new URL(info.url);
} catch (error) {
  logger.error('Invalid URL provided:', info.url);
  return false;
}
```